src/utils/index.ts

```diff
@@ -163,3 +163,29 @@ export function assertIsTruthy<T>(val: any): asserts val is NonNullable<T> {
     throw new AssertionError({ expected: true, actual: val })
   }
 }
+
+export enum IdentityNameValidationError {
+  MINIMUM_LENGTH = 'error_minimum_length',
+  MAXIMUM_LENGTH = 'error_maximum_length',
+  ILLEGAL_CHARACTER = 'error_illegal_character'
+}
+
+const containsLegalCharacters = (name: string) => /^[a-z0-9_]+$/.test(name)
+
+export const validateIdentityName = (identityName: string): IdentityNameValidationError | null => {
+  const nameLength = identityName.length
+
+  if (nameLength < 8) {
+    return IdentityNameValidationError.MINIMUM_LENGTH
+  }
+
+  if (nameLength > 38) {
+    return IdentityNameValidationError.MINIMUM_LENGTH
+  }
+
+  if (!containsLegalCharacters(identityName)) {
+    return IdentityNameValidationError.ILLEGAL_CHARACTER
+  }
+
+  return null
+}
```

Comment:
Should these be throwing instead of just returning?

Reply:
I did think about this, but I reckon that'll make the api used to consume it more complicated, requiring try/catch etc. I'd be more inclined to do this if it were consuming with Promising/Observables.

Reply:
but can easily switch to throwing error when consumed.

```ts
identityName$.pipe(
  map(name => {
    const errors = validateIdentityName(name);
    if (errors !== null) throw new Error(errors(); 
  }),
  catchError(error = handleError(error))
)
```